eng/Analyzers.props

```diff
@@ -1,9 +1,9 @@
 <Project>
-  <PropertyGroup>
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' != 'true'">
```

Comment:
Instead of modifying this one, if you always need to exclude analyzers, I'd exclude them here with this condition:

https://github.com/dotnet/diagnostics/blob/7f4401886482a7bdce22ce430bc304b324089402/src/Directory.Build.props#L30

Reply:
I'd also revert to `ArcadeBuildFromSource`, given that this is in `SourceBuildArcadeBuild.targets` (with this you'd need to provide both).

Reply:
Good suggestion on where to move the source-build condition.  I fixed this.

`DotNetBuildFromSource` should be all that is needed for the condition as Arcade source-build infra sets it internally.

Reply:
Looks good to me then.